nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/util/AbstractAttributesParser.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2017-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.p2.internal.util;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.PropertyResourceBundle;
+import java.util.jar.JarEntry;
+import java.util.jar.JarInputStream;
+
+import javax.annotation.Nullable;
+import javax.xml.xpath.XPathExpressionException;
+
+import org.sonatype.nexus.repository.p2.internal.exception.AttributeParsingException;
+import org.sonatype.nexus.repository.p2.internal.metadata.P2Attributes;
+import org.sonatype.nexus.repository.storage.TempBlob;
+
+/**
+ *
+ * @since 0.next
+ */
+public abstract class AbstractAttributesParser
```

Comment:
it does not looks like parent for AttributesParserFeatureXml and AttributesParserManifest